### DISCUSSION TITLE
OpenFileDialog or OpenFilePickerAsync ?

### DISCUSSION THREAD

**Discussion by pzs7602** [AvaloniaUI/Avalonia#10404] [Q&A]
In Preview 3, I use OpenFileDialog object and its ShowAsync method for users to choose a file to open, but in preview 4, the method is obsolete so I switch to use StorageProvider's OpenFilePickerAsync method. In Preview 5, something have been changed, for instance, the method  TryGetUri in IStorageFile is not available, so I have to adjust my code, which is a big job. My question is: Can I still use OpenFileDialog(which is the easy way) in Preview5 or later release?

**Comment by maxkatz6** [comment]
> so I have to adjust my code, which is a big job

It is expected from preview builds. 11.0-preview is not a stable release.

> My question is: Can I still use OpenFileDialog

You can. But StorageProvider is recommended.

> the method TryGetUri

file.Path it is now. In next preview release there will be an extension method `file.TryGetLocalPath()` too.
